### PR TITLE
Plane: Autotest for GCS Failsafe

### PR DESCRIPTION
**Pull request by kknives**
#19190 
This PR adds regression tests for GCS failsafe by testing behavior in case of telemetry loss as set by `FS_LONG_ACTN`.
The following modes have been tested:
- RTL
- FBWA Glide
- Parachute

I did not include tests for short failsafe, as according to the [wiki](https://ardupilot.org/plane/docs/apms-failsafe-function.html):
> Short failsafe only occurs on loss of RC, either RC loss or Throttle Failsafe.

Please feel free to point out any changes required as this is my first PR here.

### CODE REVIEW

File: Tools/autotest/arduplane.py [hunk 1]
```diff
@@ -1164,6 +1164,55 @@ def ThrottleFailsafeFence(self):
...
 
+    def GCSFailsafe(self):
+        '''Ensure Long-Failsafe works on GCS loss'''
+        self.start_subtest("Test Failsafe: RTL")
+        self.load_sample_mission()
+        self.set_parameter("RTL_AUTOLAND", 1)
+        self.change_mode("AUTO")
+        self.takeoff()
+        self.set_parameter("FS_GCS_ENABL", 1)
+        self.set_parameter("FS_LONG_ACTN", 1)
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py:1175]: No biggy, but there is `set_parameters` to set multiple at once

File: Tools/autotest/arduplane.py [hunk 2]
```diff
@@ -1164,6 +1164,55 @@ def ThrottleFailsafeFence(self):
...
+        self.progress("Disconnecting GCS")
+        self.set_heartbeat_rate(0)
+        self.delay_sim_time(5)
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py:1178]: Shoudln't need to `delay_sim_time` before `wait_mode`.  If you need to wait for longer there's a timeout parameter to `wait_mode`

File: Tools/autotest/arduplane.py [hunk 3]
```diff
@@ -1164,6 +1164,55 @@ def ThrottleFailsafeFence(self):
...
+
+        self.start_subtest("Test Failsafe: FBWA Glide")
+        self.set_parameter("RTL_AUTOLAND", 1)
+        self.change_mode("AUTO")
+        self.takeoff()
+        self.set_parameter("FS_GCS_ENABL", 1)
+        self.set_parameter("FS_LONG_ACTN", 2)
+        self.progress("Disconnecting GCS")
+        self.set_heartbeat_rate(0)
+        self.delay_sim_time(5)
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py:1191]: Again, don't need this delay

**Review by peterbarker — APPROVED** [review]
LGTM

File: Tools/autotest/arduplane.py (hunk 2 above)
Comment (kknives) [reply Tools/autotest/arduplane.py:1178]: Ah, yes. That makes sense, thanks for pointing that out.

File: Tools/autotest/arduplane.py (hunk 3 above)
Comment (kknives) [reply Tools/autotest/arduplane.py:1191]: I have corrected these in #23038

**Pull request merged**